reference/src/client/src/sa_engine_pkey.c

```diff
@@ -505,8 +511,14 @@ static int pkey_digestsign(
     }
 
     sa_signature_algorithm signature_algorithm = SA_SIGNATURE_ALGORITHM_EDDSA;
-    if (sa_crypto_sign(signature, signature_length, signature_algorithm, data->private_key, in, in_length,
-                NULL) != SA_STATUS_OK) {
+    sa_status status = sa_crypto_sign(signature, signature_length, signature_algorithm, data->private_key, in,
+            in_length, NULL);
+    if (status == SA_STATUS_OPERATION_NOT_SUPPORTED) {
+        ERROR("sa_crypto_sign operation not supported");
+        return -2;
```

Comment:
Recommend using an enumeration for readability and to facilitate future code changes.

Reply:
Done